### ISSUE TITLE
Do not allow to save some settings to disk while running NVDA from the installer

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#18171]
### Steps to reproduce:
* Run NVDA installer and continue with a temporary copy
* Enter new words in the dictionary or remove some of them
* Add some gestures in the input gesture dialog
* Change some options in NVDA settings dialog and save the config with `NVDA+control+c`.

### Actual behavior:
One can add new word in the dictionary and add input gesture. Only the config from NVDA settings dialog cannot be saved.
### Expected behavior:
No parameter should be saved to disk when using temporary mode as done for NVDA settings. Indeed:
1. If the way gestures are saved to disk is changed in the future (e.g. implemente layered gestures) but NVDA is finally not installed, one would end up with an invalid `gestures.ini` file.
2. The same way, dictionary file format can change in the future; it has already been so in the past. This would cause invalid format in case NVDA is finally not installed.

### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
Temp version
#### NVDA version:
2025.1beta8
#### Windows version:
Windows 10 22H2 (AMD64) build 19045.5854
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, but should be present on all previous versions.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

**Comment by CyrilleB79** [comment]
Additionally, from a temporary copy, you can:
* Create new profiles, even if you can't save any setting in it
* Delete an existing profile
* Probably rename a profile (not tested)
* Modify the profile enabled by a trigger through the triggers window

This should not be possible from the installer (in case the way it's done changes in the future. There is also no use case for this.
